client/src/panels/panel_declarations/misc/key_concept_panels/KeyConcepts.js

```diff
@@ -0,0 +1,146 @@
+import classNames from "classnames";
+import { ButtonToolbar } from "react-bootstrap";
+import MediaQuery from "react-responsive";
+import { withRouter } from "react-router";
+
+import { has_local_storage } from "src/core/feature_detection.js";
+
+import { util_components, breakpoints } from "../../shared.js";
+
+import "./KeyConcepts.scss";
+
+import common_lang from "./common_questions.yaml";
+import fin_lang from "./financial_questions.yaml";
+import ppl_lang from "./people_questions.yaml";
+import results_lang from "./results_questions.yaml";
+import tag_lang from "./tagging_questions.yaml";
+
+const {
+  create_text_maker_component,
+  AutoAccordion,
+  KeyConceptList,
+} = util_components;
+
+const { text_maker, TM } = create_text_maker_component([
+  common_lang,
+  fin_lang,
+  ppl_lang,
+  results_lang,
+  tag_lang,
+]);
+
+class KeyConcepts_ extends React.Component {
+  constructor(props) {
+    super(props);
+    this.accordionRef = React.createRef(null);
+
+    let can_pin;
+    if (has_local_storage) {
+      try {
+        can_pin = JSON.parse(localStorage.getItem(`can_pin_key_concepts`));
+      } catch {
+        can_pin = null;
+      }
+    }
+
+    this.keyConceptsContainerRef = React.createRef();
+
+    this.state = {
+      key_concepts_out_of_view: false,
+      can_pin: _.isBoolean(can_pin) ? can_pin : true,
+      key_concepts_width: null,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", () => {
+      const keyConceptsContainerTop = this.keyConceptsContainerRef.current.getBoundingClientRect()
+        .top;
+      if (keyConceptsContainerTop < 0) {
+        !this.state.key_concepts_out_of_view &&
+          this.setState({ key_concepts_out_of_view: true });
+      } else {
+        this.state.key_concepts_out_of_view &&
+          this.setState({ key_concepts_out_of_view: false });
+      }
+    });
+
+    this.setState({
+      key_concepts_width: this.keyConceptsContainerRef.current.offsetWidth,
+    });
```

Comment:
Personally, I'd move this to the end of `componentDidMount`. It's going to happen async, so it becomes unclear wether the state update has occured\* or not yet during the rest of `componentDidMount` finishes running.

\* actually, I'm not sure if React will flush a state update queue for a component before all of its running lifecycle methods are resolved? Better than raising these in-the-weeds questions is just to only make state updates at the end of a lifecycle method